pkg/pkg.bzl

```diff
@@ -132,12 +132,16 @@ def _pkg_tar_impl(ctx):
     if ctx.attr.empty_dirs:
         args += ["--empty_dir=%s" % empty_dir for empty_dir in ctx.attr.empty_dirs]
     if ctx.attr.extension and not ctx.executable.compressor:
-        dotPos = ctx.attr.extension.find(".")
-        if dotPos > 0:
-            dotPos += 1
-            args += ["--compression=%s" % ctx.attr.extension[dotPos:]]
-        elif ctx.attr.extension == "tgz":
-            args += ["--compression=gz"]
+        compression = None
+        dotPos = ctx.attr.extension.rfind(".")
+        if dotPos >= 0:
+            compression = ctx.attr.extension[dotPos+1:]
+        else:
+            compression = ctx.attr.extension
+        if compression == "tgz":
+            compression = "gz"
+        if compression and compression in SUPPORTED_TAR_COMPRESSIONS:
+            args += ["--compression=%s" % compression]
```

Comment:
Shouldn't this `fail` if an unsupported compression scheme is used?

Reply:
Whoops. Yes. Done